File: clients/ios-native/TrailNoteApp/Core/Contracts.swift
import Foundation

enum TrailRouteDifficulty: String, Codable {
    case beginner = "BEGINNER"
    case intermediate = "INTERMEDIATE"
    case advanced = "ADVANCED"
}

enum TrailViewState: String {
    case idle
    case loading
    case success
    case error
}

enum TrailRecordFlowState: String {
    case idle
    case recording
    case paused
    case ended
}

enum TrailPublishStatus: String, Codable {
    case draft = "DRAFT"
    case pendingReview = "PENDING_REVIEW"
    case published = "PUBLISHED"
    case rejected = "REJECTED"
    case archived = "ARCHIVED"
}

enum TrailInteractionAction: String {
    case toggleFavorite
    case toggleFollow
    case saveDraft
    case submitForReview
}

File: clients/android-native/app/src/main/java/com/trailnote/android/core/Contracts.kt
package com.trailnote.android.core

enum class TrailViewState {
    IDLE,
    LOADING,
    SUCCESS,
    ERROR,
}

enum class TrailRecordFlowState {
    IDLE,
    RECORDING,
    PAUSED,
    ENDED,
}

enum class TrailRouteDifficulty {
    BEGINNER,
    INTERMEDIATE,
    ADVANCED,
}

enum class TrailPublishStatus {
    DRAFT,
    PENDING_REVIEW,
    PUBLISHED,
    REJECTED,
    ARCHIVED,
}

enum class TrailInteractionAction {
    TOGGLE_FAVORITE,
    TOGGLE_FOLLOW,
    SAVE_DRAFT,
    SUBMIT_FOR_REVIEW,
}
